MAINT: wheel upload failures

cc @rgommers @mattip 

I think we've adopted the NumPy infrastructure on Azure for the most part, but perhaps it is more common for us to have to upload the same-name wheel to the "staging area" than for NumPy; perhaps we should consider adding `--force` since these are for staging only anyway?

```
/bin/bash --noprofile --norc /Users/runner/work/_temp/03c63e73-38e7-4f0b-b57b-4b29a1d074be.sh
uploading wheelhouse/scipy-1.6.1-cp37-cp37m-macosx_10_9_x86_64.whl
Using Anaconda API: https://api.anaconda.org
[DEBUG] Using token from command line args
Using "multibuild-wheels-staging" as upload username
Processing 'wheelhouse/scipy-1.6.1-cp37-cp37m-macosx_10_9_x86_64.whl'
Detecting file type...
[DEBUG] Testing if conda package ..
[DEBUG] Not conda package (file ext is not .tar.bz2)
[DEBUG] Testing if pypi package ..
[DEBUG] This is a pypi wheel package
File type is "pypi"
Extracting python package attributes for upload
Creating package "scipy"
Creating release "1.6.1"
Uploading file "multibuild-wheels-staging/scipy/1.6.1/scipy-1.6.1-cp37-cp37m-macosx_10_9_x86_64.whl"
Distribution already exists. Please use the -i/--interactive or --force or --skip options or `anaconda remove multibuild-wheels-staging/scipy/1.6.1/scipy-1.6.1-cp37-cp37m-macosx_10_9_x86_64.whl
##[error]Bash exited with code '1'.
Finishing: Upload to anaconda.org (only if secret token is retrieved)

```

--- a/azure-posix.yml
+++ b/azure-posix.yml
@@ -120,7 +120,7 @@ jobs:
 
           echo uploading wheelhouse/*.whl
 
-          anaconda -v -t $TOKEN upload -u $ANACONDA_ORG wheelhouse/*.whl
+          anaconda -v -t $TOKEN upload --force -u $ANACONDA_ORG wheelhouse/*.whl
           echo "PyPI-style index: https://pypi.anaconda.org/$ANACONDA_ORG/simple"
         displayName: Upload to anaconda.org (only if secret token is retrieved)
         condition: ne(variables['TOKEN'], '')

